src/sudoers/mod.rs

```diff
@@ -222,19 +222,19 @@ fn group_cmd_specs_per_runas<'a>(
         groups: Vec::new(),
     };
 
-    let mut runas = None;
+    let mut last_runas = None;
     let mut collected_specs = vec![];
 
-    for (new_runas, (tag, spec)) in cmnd_specs {
-        if let Some(new_runas) = new_runas {
+    for (runas, (tag, spec)) in cmnd_specs {
+        if runas.map(|r| r as *const _) != last_runas.map(|r| r as *const _) {
```

Comment:
I don't really like that this depends on the address of the string rather than the content.

Reply:
That's unavoidable; if you compare it by content (which are not strings) this code will group together things that are on different lines in /etc/sudoers. Which I would be fine with, except that ogsudo doesn't do that. I.e. the actual location of the origin of the RunAs spec matters here and then using the address (which is unique for every occurrence of a runas-spec in /etc/sudoers) makes really good sense.

The old code was doing the wrong thing by assuming that a "None" here signified a continuation of the old RunAs spec (which, if it didn't, would have immediately make us notice #974)

Reply:
So actually I think it's pretty elegant and a very neat and safe way to use pointers for a change :-)

Reply:
Of course a comment to this affect at this place in the code would be in order.

Reply:
Would you mind adding a comment to that extend?

Reply:
Note that another solution would be to add a "source identifier" to `RunAs`, of course, but that's more work than just using a pointer :-P

Reply:
> Would you mind adding a comment to that extend?

Definitely. Once I figure out why my rewrite of `group_cmnd_..` is failing (or until I give up on that commit)